v2/internal/loader/writing_loader.go

```diff
@@ -46,25 +49,50 @@ type IndexTracker interface {
 	Index() (index.Index, error)
 }
 
+var _ IndexTracker = (*indexingWriter)(nil)
+
 type writingReader struct {
 	r   io.Reader
-	len int64
+	buf []byte
 	cid string
-	wo  *writerOutput
+	wo  *indexingWriter
```

Comment:
maybe rename this so it's not got an unexplainable name for future eyes?

```suggestion
	iw  *indexingWriter
```